subsys/bluetooth/host/l2cap.c

```diff
@@ -70,47 +70,19 @@ NET_BUF_POOL_FIXED_DEFINE(disc_pool, 1,
 #define l2cap_lookup_ident(conn, ident) __l2cap_lookup_ident(conn, ident, false)
 #define l2cap_remove_ident(conn, ident) __l2cap_lookup_ident(conn, ident, true)
 
-struct l2cap_tx_meta_data {
-	uint16_t cid;
-	bt_conn_tx_cb_t cb;
-	void *user_data;
-};
-
-struct l2cap_tx_meta {
-	struct l2cap_tx_meta_data *data;
-};
-
-static struct l2cap_tx_meta_data l2cap_tx_meta_data_storage[CONFIG_BT_CONN_TX_MAX];
-K_FIFO_DEFINE(free_l2cap_tx_meta_data);
-
-static struct l2cap_tx_meta_data *alloc_tx_meta_data(void)
+static void *tx_cid_to_user_data(struct bt_l2cap_le_chan *chan)
 {
-	return k_fifo_get(&free_l2cap_tx_meta_data, K_NO_WAIT);
+	/* trust me bro */
+	return (void *)((uint32_t)chan->tx.cid);
```

Comment:
Zephyr has a macro for this for some reason: https://docs.zephyrproject.org/latest/kernel/util/index.html#c.UINT_TO_POINTER Not sure if there is any point in using it here though, since we anyway have to cast to 32 bits.

Reply:
I'll use it, as void* is 64b on native_64